### PR TITLE
Add Snapshot Generation Script

### LINKED ISSUE #609: [Feature] Script to Generate Snapshots for Supported SQL Engines
### Is this your first time submitting a feature request?

- [X] I have read the [expectations for open source contributors](https://docs.getdbt.com/docs/contributing/oss-expectations)
- [X] I have searched the existing issues, and I could not find an existing issue for this feature
- [X] I am requesting a straightforward extension of existing metricflow functionality, rather than a Big Idea better suited to a discussion

### Describe the feature

Test snapshots occasionally need to be regenerated (e.g. due to a naming change). Some of these snapshots need to be generated for each supported SQL engine. A script to do this would improve the developer experience.

### Describe alternatives you've considered

_No response_

### Who will this benefit?

_No response_

### Are you interested in contributing this feature?

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by plypaul**
Resolves #609

### Description

Implements the feature described in the linked issue.

<!--- 
  Before requesting review, please make sure you have:
  1. read [the contributing guide](https://github.com/dbt-labs/metricflow/blob/main/CONTRIBUTING.md),
  2. signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
  3. run `changie new` to [create a changelog entry](https://github.com/dbt-labs/metricflow/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: metricflow/test/generate_snapshots.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    @property
+    def as_sequence(self) -> Sequence[MetricFlowTestCredentialSet]:  # noqa: D
+        return (self.duck_db, self.postgres, self.redshift, self.snowflake, self.big_query, self.databricks)
+
+
+SNAPSHOT_GENERATING_TEST_FILES = (
+    "metricflow/test/dataflow/builder/test_dataflow_plan_builder.py",
+    "metricflow/test/plan_conversion/test_dataflow_to_sql_plan.py",
+    "metricflow/test/integration/test_rendered_query.py",
+)
```
Comment (WilliamDee) [inline metricflow/test/generate_snapshots.py:70]: There's also tests here `metricflow/test/dataflow/optimizer/source_scan/` that have snapshot generating tests

Comment (plypaul) [reply metricflow/test/generate_snapshots.py:70]: Added.

Comment (tlento) [reply metricflow/test/generate_snapshots.py:72]: https://github.com/dbt-labs/metricflow/blob/main/metricflow/test/model/test_data_warehouse_tasks.py#L229-L253

There are others missing too. Even my old -k filter (`pytest -k "dataflow or plan or optimizer or source_scan or test_rendered_query or test_build_metric_tasks" ...`) doesn't cover it all anymore.

Comment (WilliamDee) [reply metricflow/test/generate_snapshots.py:72]: Screenshot 2023-06-20 at 2 26 19 PM
Screenshot 2023-06-20 at 2 27 15 PM
seems to be more actually with 
```
- test_rendered_query.py
- test_data_warehouse_tasks.py
- compare_sql_plan.py
```

File: metricflow/test/generate_snapshots.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    def as_sequence(self) -> Sequence[MetricFlowTestCredentialSet]:  # noqa: D
+        return (self.duck_db, self.postgres, self.redshift, self.snowflake, self.big_query, self.databricks)
+
+
+SNAPSHOT_GENERATING_TEST_FILES = (
+    "metricflow/test/dataflow/builder/test_dataflow_plan_builder.py",
+    "metricflow/test/dataflow/optimizer/source_scan/test_cm_branch_combiner.py",
+    "metricflow/test/dataflow/optimizer/source_scan/test_source_scan_optimizer.py",
+    "metricflow/test/integration/test_rendered_query.py",
+    "metricflow/test/plan_conversion/test_dataflow_to_sql_plan.py",
```
Comment (tlento) [inline metricflow/test/generate_snapshots.py:71]: I think this is the complete set, but we won't know until we try it.

```
$ git grep -l "assert.*snapshot" | sort
metricflow/test/dataflow/builder/test_dataflow_plan_builder.py
metricflow/test/dataflow/optimizer/source_scan/test_cm_branch_combiner.py
metricflow/test/dataflow/optimizer/source_scan/test_source_scan_optimizer.py
metricflow/test/integration/test_rendered_query.py
metricflow/test/model/test_data_warehouse_tasks.py
metricflow/test/plan_conversion/test_dataflow_to_sql_plan.py
metricflow/test/plan_utils.py
metricflow/test/sql/compare_sql_plan.py
metricflow/test/table_snapshot/test_table_snapshots.py
```

**Review by tlento — APPROVED** [review]
Neat. You're still missing some paths but let's get this in once you've updated and verified they work. Even if they don't this is easy to modify.

I'm going to have to fork all of the engine-specific test runs into separate hatch environments, and at that point we might want to parameterize make test-<engine> calls, but centralizing these things is really nice.

File: metricflow/test/generate_snapshots.py (hunk 2 above)
Comment (plypaul) [reply metricflow/test/generate_snapshots.py]: Updated.

**Pull request merged**